src/authn/Token.ts

```diff
@@ -0,0 +1,38 @@
+import { authn } from '@xmtp/proto'
+import AuthData from './AuthData'
+
+export default class Token implements authn.Token {
+  identityKey: authn.PublicKey
+  authDataBytes: Uint8Array
+  authDataSignature: authn.Signature
+  private _authData?: AuthData
+
+  constructor({ identityKey, authDataBytes, authDataSignature }: authn.Token) {
+    if (!identityKey) {
+      throw new Error('Missing identity key in token')
```

Comment:
If the key signature is optional, we should check that it is indeed signed as well.